object_manager_mx_record.go

```diff
@@ -41,47 +45,59 @@ func (objMgr *ObjectManager) GetMXRecordByRef(ref string) (*RecordMX, error) {
 	return recordMX, err
 }
 
-func (objMgr *ObjectManager) GetMXRecord(dnsview string, fqdn string) (*RecordMX, error) {
-	if dnsview == "" || fqdn == "" {
+func (objMgr *ObjectManager) GetMXRecord(dnsView string, fqdn string) (*RecordMX, error) {
+	if dnsView == "" || fqdn == "" {
 		return nil, fmt.Errorf("DNS view and fqdn are required to retrieve a unique mx record")
 	}
 	var res []RecordMX
 
 	recordMX := NewRecordMX(RecordMX{})
 
 	sf := map[string]string{
-		"view": dnsview,
+		"view": dnsView,
 		"name": fqdn,
 	}
 	queryParams := NewQueryParams(false, sf)
 	err := objMgr.connector.GetObject(recordMX, "", queryParams, &res)
 
 	if err != nil {
 		return nil, err
-	} else if res == nil || len(res) == 0 {
-		return nil, NewNotFoundError(
-			fmt.Sprintf(
-				"MX record with name '%s' in DNS view '%s' is not found",
-				fqdn, dnsview))
 	}
 
 	return &res[0], err
 }
 
 func (objMgr *ObjectManager) UpdateMXRecord(
 	ref string,
-	dnsview string,
+	dnsView string,
 	fqdn string,
 	mx string,
+	ttl uint32,
+	useTtl bool,
 	comment string,
 	priority int,
 	eas EA) (*RecordMX, error) {
 
+	res, _ := objMgr.GetMXRecordByRef(ref)
```

Comment:
Why the error is ignored here? This is quite an important place to stop if something bad happens.